modules/Slider.js

```diff
@@ -0,0 +1,214 @@
+try { // For making it possiblie to run the test app in the following catch statement. It would complain on `exports` not being defined when uploading from Web IDE and not run otherwise.
+
+  exports.create = function(cb, conf) {
+
+    const R = Bangle.appRect;
+
+    // Empty function added to cb if it's undefined.
+    if (!cb) cb = ()=>{};
+
+    let o = {};
+    o.v = {}; // variables go here.
+    o.f = {}; // functions go here.
+
+    // Default configuration for the indicator, modified by parameter `conf`:
+    o.c = Object.assign({ // constants go here.
+      currLevel:undefined,
+      horizontal:false,
+      xStart:R.x2-R.w/4-4,
+      width:R.w/4,
+      yStart:R.y+4,
+      height:R.h-10,
+      steps:30,
+
+      dragableSlider:true,
+      dragRect:R,
+      useIncr:true,
+      useMap:false,
+      oversizeR:0,
+      oversizeL:0,
+      propagateDrag:false,
+      timeout:1,
+
+      drawableSlider:true,
+      colorFG:g.theme.fg2,
+      colorBG:g.theme.bg2,
+      rounded:true,
+      outerBorderSize:2,
+      innerBorderSize:2,
+
+      autoProgress:false,
+    },conf);
+
+    // If horizontal, flip things around.
+    if (o.c.horizontal) {
+      let mediator = o.c.xStart;
+      o.c.xStart = o.c.yStart;
+      o.c.yStart = mediator;
+      mediator = o.c.width;
+      o.c.width = o.c.height;
+      o.c.height = mediator;
+      delete mediator;
+    }
+
+    // Make room for the border. Underscore indicates the area for the actual indicator bar without borders.
+    let totalBorderSize = o.c.outerBorderSize + o.c.innerBorderSize;
+    o.c._xStart = o.c.xStart + totalBorderSize;
+    o.c._width = o.c.width - 2*totalBorderSize;
+    o.c._yStart = o.c.yStart + totalBorderSize;
+    o.c._height = o.c.height - 2*totalBorderSize;
+    o.c.rounded = o.c.rounded?20:0;
+
+    o.c.STEP_SIZE = ((!o.c.horizontal?o.c._height:o.c._width)-(!o.c.rounded?0:(2*o.c.rounded-7)))/o.c.steps;
+
+    // Add a rectangle object with x, y, x2, y2, w and h values.
+    o.c.r = {x:o.c.xStart, y:o.c.yStart, x2:o.c.xStart+o.c.width, y2:o.c.yStart+o.c.height, w:o.c.width, h:o.c.height};
+
+    // Initialize the level
+    o.v.level = (o.c.currLevel||o.c.currLevel===0)?o.c.currLevel:o.c.steps/2;
```

Comment:
Shouldn't it rather be checking for `currLevel` in the parameter object `conf`? Because as it's set up now `currLevel` is in `o.c` even if the user doesn't pass it in (this could be changed to not be the case, as well of course).

Reply:
Yeah that's a good point, I forgot that `currLevel` will always be [in `o.c`](https://github.com/espruino/BangleApps/blob/1d5d4208999cd8e5f60328ae605841f2c9098962/modules/Slider.js#L16). I suppose we could either check if it's in `conf` like you say, or check if it's `undefined`, as in:

```suggestion
    o.v.level = (o.c.currLevel!=null?o.c.currLevel:o.c.steps/2;
```